fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -124,12 +127,19 @@
             // hasNext is false to avoid the NoNextReason changing.
             return CompletableFuture.completedFuture(nextResult);
         }
-        if (topDocs == null) {
+        if (topDocs == null || (topDocs.scoreDocs.length-1 < currentPosition && limitRemaining > 0 && !exhausted)) {
+            long startTime = System.nanoTime();
             try {
                 performScan();
-            } catch (IOException ioe) {
-                throw new RuntimeException(ioe);
+            } catch (IOException ioException) {
+                throw new RuntimeException(ioException);
             }
+            if (timer != null) {
+                timer.recordSinceNanoTime(FDBStoreTimer.Events.LUCENE_INDEX_SCAN, startTime);
+                timer.increment(FDBStoreTimer.Counts.LUCENE_SCAN_COUNT);
```

Comment:
There is no need for this third metric; `LUCENE_INDEX_SCAN` will have both a count and a total time.